### ISSUE TITLE
[0.7.0] Authorization blocked on many websites when proxy in use

### ISSUE THREAD

**Issue by mcjcloud** [hatoo/http-mitm-proxy#28]
Found a new issue when testing the latest 0.7.0 release.

The https example works as expected now. However, when the proxy is in use, I can no longer authenticate to Github or Gmail.

Even if already signed in, the browser will treat you as being signed out when the proxy is switched on.

https://github.com/hatoo/http-mitm-proxy/assets/14041091/c786120a-40b2-49ea-9273-5605884f9ab2

EDIT: This is only on the `dev_proxy` example. The `proxy` example works as expected.

**Comment by mcjcloud** [comment]
I was able to fix it locally by replacing this

```rust
let proxy = MitmProxy::new(
        // This is the root cert that will be used to sign the fake certificates
        Some(root_cert),
        // This is the connector that will be used to connect to the upstream server from proxy
        tokio_native_tls::native_tls::TlsConnector::new().unwrap(),
);
```

with this

```rust
let proxy = MitmProxy::new(
        // This is the root cert that will be used to sign the fake certificates
        Some(root_cert),
        // This is the connector that will be used to connect to the upstream server from proxy
        tokio_native_tls::native_tls::TlsConnector::builder().request_alpns(&["h2", "http/1.1"]).build().unwrap(),
);
```
